### PR TITLE
Converting to structured logging the file store/sqlstore/upgrade.go

### PR DESCRIPTION
**Pull request by niklabh**
#### Summary

Converting to structured logging the file store/sqlstore/upgrade.go

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/12055

### CODE REVIEW

**Comment by lindalumitchell** [comment]
QA not needed on this PR; removing QA Review label.

File: store/sqlstore/upgrade.go [hunk 1]
```diff
@@ -170,17 +169,17 @@ func UpgradeDatabase(sqlStore SqlStore, currentModelVersionString string) error
 
 func saveSchemaVersion(sqlStore SqlStore, version string) {
 	if err := sqlStore.System().SaveOrUpdate(&model.System{Name: "Version", Value: version}); err != nil {
-		mlog.Critical(err.Error())
+		mlog.Critical("error", mlog.Err(err))
```
Comment (cpanato) [inline store/sqlstore/upgrade.go]: maybe chang the `error` to `critical`

File: store/sqlstore/upgrade.go [hunk 2]
```diff
@@ -478,7 +477,7 @@ func UpgradeDatabaseToVersion49(sqlStore SqlStore) {
 		defaultTimezone := timezones.DefaultUserTimezone()
 		defaultTimezoneValue, err := json.Marshal(defaultTimezone)
 		if err != nil {
-			mlog.Critical(fmt.Sprint(err))
+			mlog.Critical("error", mlog.Err(err))
```
Comment (cpanato) [inline store/sqlstore/upgrade.go]: same here

File: store/sqlstore/upgrade.go (hunk 1 above)
Comment (niklabh) [reply store/sqlstore/upgrade.go]: done

File: store/sqlstore/upgrade.go (hunk 2 above)
Comment (niklabh) [reply store/sqlstore/upgrade.go]: done

**Comment by deanwhillier** [comment]
Asking @mkraft to review instead of me, thx @mkraft!

File: store/sqlstore/upgrade.go [hunk 3]
```diff
@@ -118,7 +117,7 @@ func UpgradeDatabase(sqlStore SqlStore, currentModelVersionString string) error
 	if currentSchemaVersion.GTE(nextUnsupportedMajorVersion) {
 		return errors.Errorf("Database schema version %s is not supported. This Mattermost server supports only >=%s, <%s. Please upgrade to at least version %s before continuing.", *currentSchemaVersion, currentModelVersion, nextUnsupportedMajorVersion, nextUnsupportedMajorVersion)
 	} else if currentSchemaVersion.GT(currentModelVersion) {
-		mlog.Warn(fmt.Sprintf("The database schema with version %s is newer than Mattermost version %s.", currentSchemaVersion, currentModelVersion))
+		mlog.Warn("The database schema with version ", mlog.String("version", currentSchemaVersion.String()), mlog.String("is newer than Mattermost version", currentModelVersion.String()))
```
Comment (mkraft) [inline store/sqlstore/upgrade.go]:
```suggestion
		mlog.Warn("The database schema version and model versions do not match", mlog.String("schema_version", currentSchemaVersion.String()), mlog.String("model_version", currentModelVersion.String()))
```

File: store/sqlstore/upgrade.go [hunk 4]
```diff
@@ -170,17 +169,17 @@ func UpgradeDatabase(sqlStore SqlStore, currentModelVersionString string) error
 
 func saveSchemaVersion(sqlStore SqlStore, version string) {
 	if err := sqlStore.System().SaveOrUpdate(&model.System{Name: "Version", Value: version}); err != nil {
-		mlog.Critical(err.Error())
+		mlog.Critical("critical", mlog.Err(err))
```
Comment (mkraft) [inline store/sqlstore/upgrade.go]: I'm just questioning if this change adds value... what is the thinking here?

File: store/sqlstore/upgrade.go [hunk 5]
```diff
@@ -170,17 +169,17 @@ func UpgradeDatabase(sqlStore SqlStore, currentModelVersionString string) error
...
 		time.Sleep(time.Second)
 		os.Exit(EXIT_VERSION_SAVE)
 	}
 
-	mlog.Warn(fmt.Sprintf("The database schema has been upgraded to version %v", version))
+	mlog.Warn("The database schema has been upgraded to", mlog.String("version", version))
```
Comment (mkraft) [inline store/sqlstore/upgrade.go]:
```suggestion
	mlog.Warn("The database schema version has been upgraded", mlog.String("version", version))
```

File: store/sqlstore/upgrade.go [hunk 6]
```diff
@@ -105,7 +104,7 @@ func UpgradeDatabase(sqlStore SqlStore, currentModelVersionString string) error
 		}
 
 		currentSchemaVersion = &currentModelVersion
-		mlog.Info(fmt.Sprintf("The database schema has been set to version %s", *currentSchemaVersion))
+		mlog.Info("The database schema has been set to", mlog.String("version", currentSchemaVersion.String()))
```
Comment (mkraft) [inline store/sqlstore/upgrade.go]:
```suggestion
		mlog.Info("The database schema version has been set", mlog.String("version", currentSchemaVersion.String()))
```

File: store/sqlstore/upgrade.go [hunk 7]
```diff
@@ -170,17 +169,17 @@ func UpgradeDatabase(sqlStore SqlStore, currentModelVersionString string) error
...
 	}
 
-	mlog.Warn(fmt.Sprintf("The database schema has been upgraded to version %v", version))
+	mlog.Warn("The database schema has been upgraded to", mlog.String("version", version))
 }
 
 func shouldPerformUpgrade(sqlStore SqlStore, currentSchemaVersion string, expectedSchemaVersion string) bool {
 	if sqlStore.GetCurrentSchemaVersion() == currentSchemaVersion {
-		mlog.Warn(fmt.Sprintf("Attempting to upgrade the database schema version from %s to %v", currentSchemaVersion, expectedSchemaVersion))
+		mlog.Warn("Attempting to upgrade the database schema version from", mlog.String("current version", currentSchemaVersion), mlog.String("to expected version", expectedSchemaVersion))
```
Comment (mkraft) [inline store/sqlstore/upgrade.go]:
```suggestion
		mlog.Warn("Attempting to upgrade the database schema version", mlog.String("current_version", currentSchemaVersion), mlog.String("new_version", expectedSchemaVersion))
```

File: store/sqlstore/upgrade.go [hunk 8]
```diff
@@ -478,7 +477,7 @@ func UpgradeDatabaseToVersion49(sqlStore SqlStore) {
 		defaultTimezone := timezones.DefaultUserTimezone()
 		defaultTimezoneValue, err := json.Marshal(defaultTimezone)
 		if err != nil {
-			mlog.Critical(fmt.Sprint(err))
+			mlog.Critical("critical", mlog.Err(err))
```
Comment (mkraft) [inline store/sqlstore/upgrade.go]: Same question here.

File: store/sqlstore/upgrade.go [hunk 9]
```diff
@@ -672,12 +671,12 @@ func UpgradeDatabaseToVersion511(sqlStore SqlStore) {
 		// Enforce all teams have an InviteID set
 		var teams []*model.Team
 		if _, err := sqlStore.GetReplica().Select(&teams, "SELECT * FROM Teams WHERE InviteId = ''"); err != nil {
-			mlog.Error("Error fetching Teams without InviteID: " + err.Error())
+			mlog.Error("Error fetching Teams without InviteID:", mlog.Err(err))
```
Comment (mkraft) [inline store/sqlstore/upgrade.go]: What about adding a `team` or `team_id` parameter to this?  Also we can remove the `:`.

File: store/sqlstore/upgrade.go [hunk 10]
```diff
@@ -672,12 +671,12 @@ func UpgradeDatabaseToVersion511(sqlStore SqlStore) {
...
 		} else {
 			for _, team := range teams {
 				team.InviteId = model.NewId()
 				if _, err := sqlStore.Team().Update(team); err != nil {
-					mlog.Error("Error updating Team InviteIDs: " + err.Error())
```
Comment (mkraft) [inline store/sqlstore/upgrade.go]: Same here, what about adding a `team` or `team_id` parameter to this? Also we can remove the `:`.

File: store/sqlstore/upgrade.go (hunk 3 above)
Comment (niklabh) [reply store/sqlstore/upgrade.go]: done

File: store/sqlstore/upgrade.go (hunk 5 above)
Comment (niklabh) [reply store/sqlstore/upgrade.go]: done

File: store/sqlstore/upgrade.go (hunk 4 above)
Comment (niklabh) [reply store/sqlstore/upgrade.go]: oh yes Sprintf is not used here no need to change reverting.

Comment (niklabh) [reply store/sqlstore/upgrade.go]: reverted

File: store/sqlstore/upgrade.go (hunk 6 above)
Comment (niklabh) [reply store/sqlstore/upgrade.go]: done

File: store/sqlstore/upgrade.go (hunk 8 above)
Comment (niklabh) [reply store/sqlstore/upgrade.go]: changed to mlog.Critical(err.Error()). Goal of the ticket is to remove Sprint formatting.

File: store/sqlstore/upgrade.go (hunk 7 above)
Comment (niklabh) [reply store/sqlstore/upgrade.go]: done

File: store/sqlstore/upgrade.go (hunk 10 above)
Comment (niklabh) [reply store/sqlstore/upgrade.go]: done

File: store/sqlstore/upgrade.go (hunk 9 above)
Comment (niklabh) [reply store/sqlstore/upgrade.go]: done

**Comment by hanzei** [comment]
@niklabh I'm wondering why this PR got closed? It seams like it was ready to be merged?

**Comment by niklabh** [comment]
I didn't closed it by hand. I think by mistake master was forced pushed and that closed it. Here is a new PR with all the review comment fixes: https://github.com/mattermost/mattermost-server/pull/12628

**Comment by niklabh** [comment]
Sorry for the inconvenience.

**Comment by hanzei** [comment]
No problem, thanks for the new PR :+1:

**Pull request closed without merging**